### ISSUE TITLE
Decouple favorites from offline availability?

### ISSUE THREAD

**Issue by tanuva** [nextcloud/ios#627]
Combining offline availability with the favorites feature looked like a pretty good idea to me at first. Now I have a folder with a few hundred MB of data. I would like to have this folder available offline on an iPad, but *not* on my iPhone. (Mostly because storage is more limited on the phone.) However, since the favorites flag is dutifully synced between my devices, the downloads also take place on both of them.

Provided you agree that having (arbitrary amounts of) files available offline is a desirable feature, I would like to propose decoupling the favorites feature from offline availability. Surely that means another button and file management code, but it would help immensely in my use case. What do you think?

I found a few other issues here where it was already suggested that favorites might not be intended for marking lots of files - still, one folder is just one folder, isn't it? :) (I think this issue is different from favoriting *too many* files.)

**iOS version:** 11.4.1
**App version:** 2.21.3.1
**Nextcloud version:** 13.0.5

**Comment by marinofaggiana** [comment]
@tanuva it's a good question,  @tobiasKaminsky in Android is the same as iOS ?

(P.S. for issue try the beta 2.22.0 on TestFlight)

**Comment by tobiasKaminsky** [comment]
On android we want to change to this behaviour:
- favorite is only a "tag" to find files/folders easier
- every 2h there is a sync job (only on wifi),which syns all downloaded files
- on entering a folder it also refreshs the files
	- on wifi: directly
	- without wifi: shows a red dot on those downloaded files that are out of sync

**Comment by TecJon** [comment]
@tanuva this definitely makes sense. I experience the same issues for the same reason. I'm glad you brought it up.

**Comment by marinofaggiana** [comment]
Thanks @tobiasKaminsky but this is not possible, the background mode in iOS is not manage from app ...

**Comment by marinofaggiana** [comment]
For example in iOS Dropbox exists 2 option for a file : 

- add to favorite
- make available offline

**Comment by TecJon** [comment]
> For example in iOS Dropbox exists 2 option for a file/folder : 
>- add to favorite
>- make available offline

I think that makes perfect sense.

**Comment by TecJon** [comment]
Any files that are marked for "make available offline" could be listed under a seperate view. This could be found in "more" below "Shares" an entry "Offline Files".

**Comment by tanuva** [comment]
Personally, I like about the current UI that there is exactly one place to browse all the available files. Therefore I would be okay with having this Offline Files entry *in addition* to the Files tab where all the files are shown, remote and downloaded. In the iOS app for the Plex Media Server, there is a separate view for artists, albums and the like that are synchronized to the device. There the download status is shown and synchronizations can also be removed again.
(I'm aware there's the Transfers view already, this is just meant as an inspiration.)

![separate view](https://45eqks401uca3p9xje465puo-wpengine.netdna-ssl.com/wp-content/uploads/sites/4/2015/05/sync_main.png)

At the same time, I think the offline files should still be accessible via the normal "Files" tab. Since there already is a marker (annotation? label?) for files that have been downloaded - the green check mark - I could imagine that becoming "two-stage". For files that were downloaded just like they are now, the green check mark stays as it is. And for files that have been explicitly made available offline there could be a new marker, let's say blue, that overrides the check mark.

**Comment by TecJon** [comment]
Yes, I agree it should definitely be _in addition_ to the normal files tab. I might have phrased that ambiguously in my comment. 
Personally, I don't see a reason to change the green check to a different check. It could be confusing to have two colors. All it does is to show that the files are downloaded and available on the client (which would be the case for files marked for offline availability). I think a green check icon would be sufficient for any downloaded file/folder. ☺️

**Comment by vomout** [comment]
If _Nextcloud-favourite = offline-available_, then we should have _Nextcloud-favourite ≠ Files-app-favourite_. However, at least in the current TestFlight version _Nextcloud-favourite = Files-app-favourite = offline-available_, which is a total disaster as I now have a list of 10 directories in Files app favourites just because I want everything available offline. I don't think this used to be so; it used to be that _offline-available = Nextcloud-favourite ≠ Files-app-favourite_. See #983. Clearest would be Nextcloud-favourite=Files-app-favourite, and offline=another flag.
